src/transformers/generation_tf_utils.py

```diff
@@ -886,6 +885,13 @@ def _generate_beam_search(
                 if (return_dict_in_generate and kwargs["encoder_hidden_states"])
                 else None
             )
+            # the refactored generate, without the encoder outputs in `past`, expects the `encoder_outputs`
+            # variable to contain all (encoder_outputs, encoder_hidden_states, encoder_attentions) in
+            # `prepare_inputs_for_generation`
+            if encoder_hidden_states is not None:
```

Comment:
Why not wrap it into a `EncoderOutputs` class here

Reply:
Great question! I tried that, it would be the most sensible change IMO (as the updated generate gets the encoder outputs with `return_dict=True`). However, a `TFEncoderOutputs` would make T5 tests fail. At this point, I had 2 options: update TF T5 or write this. Since this PR is mostly about updating the past variable, I thought it would be the path of least resistance.

Happy to change T5 instead :)